qcodes/utils/dataset/doNd.py

```diff
@@ -483,6 +483,180 @@ def do2d(
     return _handle_plotting(dataset, do_plot, interrupted())
 
 
+def dond(
+    *params: Union[_BaseParameter, int, float, ParamMeasT],
+    write_period: Optional[float] = None,
+    measurement_name: str = "",
+    exp: Optional[Experiment] = None,
+    do_plot: Optional[bool] = None,
+    show_progress: Optional[None] = None,
+    use_threads: bool = False,
+    additional_setpoints: Sequence[ParamMeasT] = tuple()
+        ) -> AxesTupleListWithDataSet:
+    """
+    Perform n-dimentional scan from slowest (first) to the fastest (last), to
+    measure m measurement parameters. Supplied params are parsed to params_set,
+    params_dicts and params_meas inside the function.
+
+    Args:
+        *params:
+            param_set_1, start_1, stop_1, num_points_1, delay_1, ...,
+            param_set_n, start_n, stop_n, num_points_n, delay_n,
+            param_meas_1, param_meas_2, ..., param_meas_m
+        write_period: The time after which the data is actually written to the
+            database.
+        measurement_name: Name of the measurement. This will be passed down to
+            the dataset produced by the measurement. If not given, a default
+            value of 'results' is used for the dataset.
+        exp: The experiment to use for this measurement.
+        do_plot: should png and pdf versions of the images be saved and plots
+            are shown after the run. If None the setting will be read from
+            ``qcodesrc.json``
+        show_progress: should a progress bar be displayed during the
+            measurement. If None the setting will be read from ``qcodesrc.json`
+        use_threads: If True, measurements from each instrument will be done on
+            separate threads. If you are measuring from several instruments
+            this may give a significant speedup.
+        additional_setpoints: A list of setpoint parameters to be registered in
+            the measurement but not scanned.
+    """
+    if do_plot is None:
+        do_plot = config.dataset.dond_plot
+    if show_progress is None:
+        show_progress = config.dataset.dond_show_progress
+
+    meas = Measurement(name=measurement_name, exp=exp)
+
+    def _make_nested_setpoints(
+        params: List[_BaseParameter],
+        param_dicts: Dict[_BaseParameter, Dict[str, Union[float, int]]]
```

Comment:
Union[float, int] is equivalent to float

Reply:
Yes,
The structure is under change. I think, it is better to review it after refactoring :)